src/Speech/SpeechClient.php

```diff
@@ -0,0 +1,494 @@
+     *           en-US (English).
+     *     @type bool $profanityFilter If set to `true`, the server will attempt
+     *           to filter out profanities, replacing all but the initial
+     *           character in each filtered word with asterisks, e.g. \"f***\".
+     *           Defaults to `false`.
+     *     @type array $speechContext Must contain a key `phrases` which is to
+     *           be an array of strings which provide "hints" to the speech
+     *           recognizer to favor specific words and phrases in the results.
+     *           Please see
+     *           [SpeechContext](https://cloud.google.com/speech/reference/rest/v1beta1/RecognitionConfig#SpeechContext)
+     *           for more information.
+     * }
+     * @return array The transcribed results. Each element of the array contains
+     *         a `transcript` key which holds the transcribed text. Optionally
+     *         a `confidence` key holding the confidence estimate ranging from
+     *         0.0 to 1.0 may be present. `confidence` is typically provided
+     *         only for the top hypothesis.
+     * @throws \InvalidArgumentException
+     */
+    public function recognize($audio, array $options = [])
+    {
+        $response = $this->connection->syncRecognize(
+            $this->formatRequest($audio, $options)
+        );
+
+        return isset($response['results']) ? $response['results'][0]['alternatives'] : [];
+    }
+
+    /**
+     * Runs a recognize request as an operation. Ideal when working with audio
+     * longer than approximately one minute. Requires polling of the returned
+     * operation in order to fetch results.
+     *
+     * The Google Cloud Client Library will attempt to infer the sample rate
+     * and encoding used by the provided audio file for you. This feature is
+     * recommended only if you are unsure of what the values may be and is
+     * currently limited to .flac, .amr, and .awb file types. The sample rate
+     * cannot be inferred from audio provided from a Google Storage object.
+     *
+     * For longer audio, up to approximately 80 minutes, you must use Google
+     * Cloud Storage objects as input. In addition to this restriction, only
+     * LINEAR16 audio encoding can be used for long audio inputs.
+     *
+     * Example:
+     * ```
+     * $operation = $speech->beginRecognizeOperation(
+     *     fopen('audio.flac', 'r')
+     * );
+     *
+     * $isComplete = $operation->isComplete();
+     *
+     * while (!$isComplete) {
+     *     sleep(1); // let's wait for a moment...
+     *     $operation->reload();
+     *     $isComplete = $operation->isComplete();
+     * }
+     *
+     * print_r($operation->results());
+     * ```
+     *
+     * ```
+     * // Run with speech context, sample rate, and encoding provided
+     * $operation = $speech->beginRecognizeOperation(
+     *     fopen('audio.flac', 'r'), [
```

Comment:
All of my examples with fopen have used an absolute path with `__DIR__`. e.g. `__DIR__ .'/audio.flac'`. It doesn't really make a difference, but we should choose one or the other.